File: frontend/src/services/feedback.ts
import api from './api';

export interface FeedbackCreate {
  project_id: string;
  section_id: string;
  liked: boolean;
}

export interface Feedback {
  id: string;
  section_id: string;
  project_id: string;
  user_id: string;
  liked: boolean;
  created_at: string;
}

export interface CommentCreate {
  project_id: string;
  section_id: string;
  comment_text: string;
}

export interface Comment {
  id: string;
  section_id: string;
  project_id: string;
  user_id: string;
  comment_text: string;
  created_at: string;
}

export const feedbackService = {
  async submitFeedback(data: FeedbackCreate): Promise<Feedback> {
    const response = await api.post<Feedback>('/feedback', data);
    return response.data;
  },

  async addComment(data: CommentCreate): Promise<Comment> {
    const response = await api.post<Comment>('/comments', data);
    return response.data;
  },
};

File: backend/app/api/v1/endpoints/feedback.py
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session
from uuid import UUID
from app.core.database import get_db
from app.api.v1.dependencies import get_current_user
from app.models.user import User
from app.models.project import Project
from app.models.section import Section
from app.models.feedback import Feedback
from app.schemas.feedback import FeedbackCreate, FeedbackResponse

router = APIRouter()

@router.post("", response_model=FeedbackResponse, status_code=status.HTTP_201_CREATED)
async def submit_feedback(
    feedback_data: FeedbackCreate,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """Submit like/dislike feedback for a section."""
    # Verify project belongs to user
    project = db.query(Project).filter(
        Project.id == feedback_data.project_id,
        Project.user_id == current_user.id
    ).first()
    
    if not project:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="Project not found"
        )
    
    # Verify section exists
    section = db.query(Section).filter(
        Section.id == feedback_data.section_id,
        Section.project_id == feedback_data.project_id
    ).first()
    
    if not section:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="Section not found"
        )
    
    # Create feedback
    feedback = Feedback(
        section_id=feedback_data.section_id,
        project_id=feedback_data.project_id,
        user_id=current_user.id,
        liked=feedback_data.liked
    )
    db.add(feedback)
    db.commit()
    db.refresh(feedback)
    
    return feedback

